### PR TITLE
Allow scoping of temporary queue prefixes

### PR DESCRIPTION
**Pull request by samesfahani**

### CODE REVIEW

File: lib/client-transport.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ function TAmqpTransport(client, _options,callback) {
     endpoint:  options.endpoint ||  'amqp://localhost',
     queueName: options.queueName || 'lycam-rpc',
     fromName: options.fromName || 'Unknown Client',
-
+    tempQueuePrefix: options.tempQueuePrefix || 'lycam-amq.gen'
```
Comment (samesfahani) [inline lib/client-transport.js]: Fun fact, you can't create a queue starting with `amq` as it is reserved (I tried), so defaulting to `lycam-amq` instead.

File: lib/client-transport.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ function TAmqpTransport(client, _options,callback) {
     endpoint:  options.endpoint ||  'amqp://localhost',
     queueName: options.queueName || 'lycam-rpc',
     fromName: options.fromName || 'Unknown Client',
-
+    tempQueuePrefix: options.tempQueuePrefix || 'lycam-amq'
```
Comment (samesfahani) [inline lib/client-transport.js:42]: Fun fact, you can't create a queue starting with amq as it is reserved (I tried), so defaulting to lycam-amq instead.

File: lib/client-transport.js [hunk 3]
```diff
@@ -180,8 +180,9 @@ TAmqpTransport.prototype.write = function (buf) {
 TAmqpTransport.prototype.flushQueue = function (msg) {
   var self = this;
   return new Promise(function (resolve, reject) {
+    var tempQueueName = self.options.tempQueuePrefix + '.gen-' + uuid.v1();
```
Comment (pater24) [inline lib/client-transport.js:183]: is there a reason why we would still do the `.gen` with uuid?

**Review by pater24 — COMMENTED** [review]
1 question

File: lib/client-transport.js (hunk 3 above)
Comment (samesfahani) [reply lib/client-transport.js:183]: Same reason amqp did it by default. Just to denote, to a human, that this queue was generated.

**Pull request merged**